src/rgw/rgw_rados.cc

```diff
@@ -3364,6 +3366,8 @@ class RGWRadosPutObj : public RGWHTTPStreamRWRequest::ReceiveCB
 
     const uint64_t lofs = data_len;
     data_len += size;
+    if (cct->_conf->rgw_copy_verify_object)
+      full_obj.append(bl);
```

Comment:
it's not reasonable to buffer the entire object for this

once we finish reading the attributes, `process_attrs()` should decode the manifest and decide which etag strategy (multipart vs. non) to use. if it's a multipart manifest, we can store the array of part sizes

for the actual etag calculation, i think an extra filter (we `rgw::putobj::DataProcessor`) would be a good way to implement that. you could derive both a `MultipartEtagFilter` and `EtagFilter` that does the streaming md5 calculation on the data passed into `process()`. `process_attrs()` would attach one of those filters before the compression step

Reply:
@cbodley - I agree that storing the entire object isn't quite efficient.

I've [re-written the patch](https://github.com/prasakri/ceph/commit/2677c4b88806d4af6d525157e7006c1b0ca1b964) to make the ETag verification as a filter based on your suggestion. Kindly review the same.